feat(icon): DLT-2553 add breakout-rooms and update coaching-hub

fede-dp opened the pull request:
# add breakout-rooms icon and update keywords

<!--- Feel free to remove any unused sections -->

## Obligatory GIF (super important!)

<!--- go to giphy.com, pick a gif, click share, copy gif link, paste within the () brackets below. -->

![lets-go-freedom-gif-by-jason-earls](https://github.com/user-attachments/assets/d52f9ee4-df2c-403a-8ff1-49ae3661bba5)

## :hammer_and_wrench: Type Of Change

<!--- Tick or place an `x` for the type of change. Should match the type in the commit message / PR title
in https://github.com/dialpad/dialtone/blob/staging/.github/COMMIT_CONVENTION.md -->

These types will increment the version number on release:

- [ ] Fix
- [x] Feature
- [ ] Performance Improvement
- [ ] Refactor

These types will not increment the version number, but will still deploy to documentation site on release:

- [ ] Documentation
- [ ] Build system
- [ ] CI
- [ ] Style (code style changes, not css changes)
- [ ] Test
- [ ] Other (chore)

## :book: Jira Ticket

<!--- Enter the URL of the Jira ticket associated with this PR -->

## :book: Description

<!--- Describe specifically what the changes are -->

## :bulb: Context

<!--- Describe the purpose of the changes -->
<!--- Why did we make these changes? -->
<!--- What problem(s) do they solve? -->

## :pencil: Checklist

<!--- Tick or place an `x` in all of the checkboxes that apply -->
<!--- Remove checkboxes that do not apply -->

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.


## :link: Sources

<!--- Add any links to external reference material -->

fede-dp committed:
feat: add breakout-rooms icon and update keywords

francisrupert commented:
@fede-dp Once you update `coaching-hub`, change the title of this PR to...
```
feat(icon): NO-JIRA added breakout-rooms and updated coaching-hub
```

Although, they probably should have an associated Jira issue.

fede-dp committed:
fix: update coaching-hub icon path for improved rendering

fede-dp commented:
> @fede-dp Once you update `coaching-hub`, change the title of this PR to...
> 
> ```
> feat(icon): NO-JIRA added breakout-rooms and updated coaching-hub
> ```
> 
> Although, they probably should have an associated Jira issue.

create and added!

juliodialpad reviewed (approved):
Thanks!

fede-dp commented:
adding the coaching hub icon from ticket [DP-141354](https://dialpad.atlassian.net/browse/DP-141354) here

[DP-141354]: https://dialpad.atlassian.net/browse/DP-141354?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

fede-dp committed:
chore: update coaching-hub icon for improved consistency

fede-dp committed:
apply the color token because isn't changing in darkmode?

braddialpad reviewed (approved):
Looks good!

francisrupert commented on packages/dialtone-icons/src/svg/icons/brand/coaching-hub.svg:
```diff
@@ -1,3 +1,3 @@
 <svg width="12" height="12" viewBox="0 0 12 12" fill="none" xmlns="http://www.w3.org/2000/svg">
-<path fill-rule="evenodd" clip-rule="evenodd" d="M3.00024 1.5C3.00024 0.947715 3.44796 0.5 4.00024 0.5H7.00024C7.55253 0.5 8.00024 0.947715 8.00024 1.5H8.50024C8.89807 1.5 9.2796 1.65804 9.5609 1.93934C9.84221 2.22064 10.0002 2.60217 10.0002 3V4C10.0002 4.27614 9.77639 4.5 9.50024 4.5C9.2241 4.5 9.00024 4.27614 9.00024 4V3C9.00024 2.86739 8.94757 2.74022 8.8538 2.64645C8.76003 2.55268 8.63285 2.5 ...
+<path d="M8.99634 11V10.5H8.49634C8.2204 10.4998 7.99634 10.276 7.99634 10C7.99639 9.72405 8.22043 9.50024 8.49634 9.5H8.99634V9C8.99639 8.72405 9.22043 8.50024 9.49634 8.5C9.77245 8.5 9.99628 8.7239 9.99634 9V9.5H10.5002L10.6008 9.51074C10.8287 9.55733 11.0002 9.75839 11.0002 10C11.0002 10.2417 10.8288 10.4437 10.6008 10.4902L10.5002 10.5H9.99634V11C9.99634 11.2761 9.77248 11.5 9.49634 11.5C9.22 ...
```
<img width="762" alt="image" src="https://github.com/user-attachments/assets/ba0c4675-34a0-467b-ab1b-6f8b0be5937a" />

Icons don't need design tokens applied here right? There's a script that scrubs that.

francisrupert replied:
In this screenshot case, you can see `coaching-hub` is brighter because it's not inheriting the color (i.e. `currentColor`)

fede-dp replied:
not sure why this wasn't changing in dark mode, chage it back to the original color

fede-dp committed:
chore: update coaching-hub icon for improved consistency

francisrupert reviewed (approved):
Good now! Note that I changed the PR title. In particular, the important part is `feat(icon)`. This ensures the release notes show the scope. 

For example, in the screenshot below...
```
NO-JIRA dialbot icon gradients rebrand 2025
```

...should have been something like...

```
fix(icon): DLT-0000 dialbot gradient updated for rebrand 2025
```

<img width="645" alt="image" src="https://github.com/user-attachments/assets/4df8271a-575d-4110-8409-485ec3293da5" />
